fontforge/splineutil2.c

```diff
@@ -3654,12 +3654,12 @@ void SplinePointListSet(SplinePointList *tobase, SplinePointList *frombase) {
 	    tsp->noprevcp = fsp->noprevcp;
 	    tsp->nonextcp = fsp->nonextcp;
 	    if ( first==NULL ) first = tsp;
-	    if ( tsp->next==NULL )
+	    if ( tsp->next==NULL || fsp->next==NULL )
 	break;
 	}
 	firstsp = NULL;
 	for ( tspline = tobase->first->next, fspline=frombase->first->next;
-		tspline!=NULL && tspline!=firstsp;
+		tspline!=NULL && frombase->spiro_cnt==0 && tspline!=firstsp;
 		tspline=tspline->to->next, fspline=fspline->to->next ) {
```

Comment:
This prevents the whole block from being executed @frank-trampe if there are Spiros as the Spiro copy below does the job. Without it we get crash:

```
Program received signal SIGSEGV, Segmentation fault.
0x0000155555105095 in SplinePointListSet (tobase=0x555555d5ac80, frombase=0x555555dfc230) at ../fontforge/splineutil2.c:3664
3664		   tspline->splines[0] = fspline->splines[0];
```

`fspline` is NULL!